File: frontend/src/types/attendance.ts
export type AttendanceStatus = 'present' | 'absent' | 'half-day' | 'leave';

export interface AttendanceRecord {
  id: string;
  employeeId: string;
  employee_id?: string;
  employeeName?: string;
  employeeAvatar?: string;
  date: string; // YYYY-MM-DD
  checkIn: string | null; // ISO timestamp or HH:mm
  check_in?: string | null;
  checkOut: string | null; // ISO timestamp or HH:mm
  check_out?: string | null;
  status: AttendanceStatus;
  workHours?: number;
  location?: string;
  notes?: string;
}

File: backend/models/attendance.py
from pydantic import BaseModel
from typing import Optional, Literal
from datetime import date, datetime

class AttendanceCreate(BaseModel):
    employee_id: Optional[str] = None
    date: Optional[date] = None
    status: Literal["present", "absent", "half-day", "leave"] = "present"

class AttendanceOut(BaseModel):
    id: str
    employee_id: str
    date: date
    check_in: Optional[datetime] = None
    check_out: Optional[datetime] = None
    status: str
    created_at: datetime

    class Config:
        from_attributes = True